app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -124,12 +124,24 @@ def analyze
                           msg % [options[:image_full_name], options[:docker_image_id][0..11], actual[0..11]],
                           'error')
     end
+
+    collect_compliance_data(image)
+
     image.scan_metadata(SCAN_CATEGORIES,
                         "taskid" => jobid,
                         "host"   => MiqServer.my_server,
                         "args"   => [YAML.dump(scan_args)])
   end
 
+  def collect_compliance_data(image)
+    # TODO: Un comment this code once an image_inspector_client that knows to fetch_oscap_arf arrives.
+    # image.openscap_result.try(:destroy)
+    # openscap_result = OpenscapResult.new(:container_image => image)
+    # openscap_result.raw = image_inspector_client.fetch_oscap_arf
```

Comment:
The order in which I delete and add is not right since I will lose old if there is a problem building new. Is this better? @jrafanie @durandom 
```
def collect_compliance_data(image)
    return unless image_inspector_client.respond_to? :fetch_oscap_arf
    openscap_result = OpenscapResult.new(:container_image => image)
    openscap_result.raw = image_inspector_client.fetch_oscap_arf
    image.openscap_result.try(:destroy) # delete old
    openscap_result.save
end
```